nemo/collections/llm/inference/base.py

```diff
@@ -130,6 +131,12 @@ def _setup_trainer_and_restore_model(
     """
     assert isinstance(trainer.strategy, MegatronStrategy), "Only MegatronStrategy is supported for trainer.strategy."
     assert trainer.strategy.context_parallel_size <= 1, "Context parallelism is not supported for inference."
+
+    # [ModelOpt]: If modelopt_state exists, overwrite transformer_layer_spec to modelopt spec
+    from nemo.collections.llm.modelopt import set_modelopt_spec_if_exists_in_ckpt
+
+    set_modelopt_spec_if_exists_in_ckpt(model, path)
```

Comment:
This allows us to automatically use modelopt spec if restoring a ckpt that has modelopt_state in it (example use case - `generate.py` on a ptq nemo ckpt)